common/src/main/java/io/druid/common/guava/GuavaUtils.java

```diff
@@ -109,4 +110,29 @@ public static Long tryParseLong(@Nullable String string)
            ? null
            : Longs.tryParse(string.charAt(0) == '+' ? string.substring(1) : string);
   }
+
+  /**
+   * Like Guava's Enums.getIfPresent, with some differences.
+   * <ul>
+   * <li>Returns nullable rather than Optional</li>
+   * <li>Permits nullable value (and returns null in this case)</li>
+   * <li>Does not require Guava 12</li>
+   * </ul>
+   */
+  @Nullable
+  public static <T extends Enum<T>> T getEnumIfPresent(final Class<T> enumClass, @Nullable final String value)
+  {
+    Preconditions.checkNotNull(enumClass, "enumClass");
+
+    if (value == null) {
+      return null;
+    }
+
+    try {
```

Comment:
Suggested to iterate over `enumClass.getEnumConstants()` to avoid operating via exception handling.